module/VuFind/src/VuFind/Service/ReCaptchaFactory.php

```diff
@@ -63,22 +63,16 @@ public function __invoke(ContainerInterface $container, $requestedName,
         }
         $config = $container->get(\VuFind\Config\PluginManager::class)
             ->get('config');
-        $siteKey = isset($config->Captcha->siteKey)
-            ? $config->Captcha->siteKey
-            : (isset($config->Captcha->publicKey)
-                ? $config->Captcha->publicKey
-                : '');
-        $secretKey = isset($config->Captcha->secretKey)
-            ? $config->Captcha->secretKey
-            : (isset($config->Captcha->privateKey)
-                ? $config->Captcha->privateKey
-                : '');
+        $siteKey = $config->Captcha->recaptcha_siteKey ??
```

Comment:
For backward compatibility, we should accept all of the old, non-prefixed configuration setting names as well (we could issue a deprecation notice if you like). We should also update the VuFind\Config\Upgrade class to adjust these as needed. (I can help with that detail if you like).

Reply:
I tried to implement this myself, please have a look

Reply:
done